### PR TITLE
Remove Bristlecone and Foxtail deprecated XmonDevices.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Before removing XmonDevice outright, remove child classes.

### CODE REVIEW

**Comment by mpharrigan** [comment]
This raises some (imo) important JSON compatibility questions. #4321 

As written, this PR drops support for a `cirq_type` completely and deletes the json test data (including the "inward" files). How hard would it be to include a shim to load the old data into a new GridDevice?

**Comment by mpharrigan** [comment]
although I see only the "old" files actually stored the qubit info; the others just give the device name. I don't know where/when that change was made.

**Comment by mpharrigan** [comment]
Proposal: keep the json resolver but have it load into like the string name of the device or something; issue a warning telling users that these named devices no longer exist as of cirq version 0.16

**or**

port the named devices to GridDevice so that they still load in a devices with qubits and such and leave them be, as they are not hurting anyone and may be in some json documents.

**Comment by MichaelBroughton** [comment]
Went with your proposal. PTAL.

File: cirq-google/cirq_google/json_resolver_cache.py [hunk 1]
```diff
@@ -20,11 +21,19 @@
...
 @functools.lru_cache()
 def _class_resolver_dictionary() -> Dict[str, ObjectFactory]:
+    def _old_xmon(*args, **kwargs):
+        warnings.warn(
+            'Attempted to json load a Bristlecone or Foxtail Device.'
+            'These devices were removed in Cirq v0.15 and are no '
+            'longer supported. Please update device usage to a '
+            'supported device or downgrade your Cirq installation.'
+        )
+        return None
```
Comment (mpharrigan) [inline cirq-google/cirq_google/json_resolver_cache.py:31]: I was suggesting we return the name or something. I think if we're at the point of returning None, we should just be raising an exception. Also: the error message can tell you what the name of the device is that failed to load.

Comment (MichaelBroughton) [reply cirq-google/cirq_google/json_resolver_cache.py]: Updated to return the name and added more info to the warning.

**Review by mpharrigan — APPROVED** [review]
LGTM -- the `json_inward` files are left unchanged, so that's good

**Pull request merged**